Add notes for Nov 2020 (#90)
tc39/tc39-notes: meetings/2020-11/nov-17.md

GCL: Yeah, I don't know if Bradley is on the call, but he brought this up in the context of certain environmental security constraints or robustness constraints. So within that context I thought it was worth making the change but I can't like speak to the specific invariants that he was working with because

SYG:I think it's fine. Like I don't disagree that it seems better to not delegate this case. I'm just trying to weigh is it worth it for us to change the thing that everybody already does the same way? I'm not really worried about the realm of the typeError itself. That is probably not going to be too much of a worry. That's it for me.

YSV: look at this change. It looks like the change will actually be an improvement for things like iterator helpers and that it will remove some hooks as long as this web compatible. I don't know if It's a huge issue to make this change. And it may be a really good simplification from my perspective.

GCL: It's cool to hear. I'd be interested to know how that affects iterator helpers,. Good to hear. 

DE: [unintelligible]

> JHD: It's always really frustrating when the use of shiny convenient syntax brings with it a whole new vulnerability to prototypes being modified. Like for example node has I mean and this is node so it can it could always find other solutions but node has this pattern internally called primordial is where it like pulls in a fresh copy, you know pre-prepared frozen copy of prototype methods and but it's use of spreads syntax is always vulnerable to user code breaking node itself and it's really painful to try and rewrite things to not use things like spread syntax to avoid this problem. And you know, like I said, node has other alternatives because it's an engine but the same sort of thing happens in user code and like babel output and so on. It's a lot that has to be worked around. So this is a wonderful change.

Reviewer: ```suggestion
JHD: It's always really frustrating when the use of shiny convenient syntax brings with it a whole new vulnerability to prototypes being modified. Like for example node has (this is node, so it can it could always find other solutions) this pattern internally called "primordials" where it like pulls in a fresh pre-frozen copy of prototype methods, but its use of spread syntax is always vulnerable to user code breaking node itself and it's really painful to try and rewrite things to not use things like spread syntax to avoid this problem. As I said, node has other alternatives because it's an engine, but the same sort of thing happens in user code and like babel output and so on. It's a lot that has to be worked around. So this is a wonderful change.
```

> JHD: I  don't actually agree with one of the things on your slide show. I don't think we have to worry too much about someone making a package that does the bad thing. I think they'll do that anyway if they want to; and they probably won't regardless, but that said, I think it looks like there's more risks than having it that aren't on this slide. As you mentioned, people will reach for `.slice(-1)[0]`, or people will have to manually compute the index that they want. I think that's a risk. That's gross code that's easy to screw up that people do all over the place. Gross is of course subjective, but I hope that that's not a contentious subjective opinion. So I think it's really important to have it. Code points aren't enough to solve what people actually want; which is some form of grapheme clusters. I think that there's a very large amount of strings that aren't human sentences where surrogate pairs matter. There's all sorts of things like code enums, and plenty of ASCII strings in which code unit indexing is exactly what's desired, which is why the “slice” and “length minus 1” patterns are common. So I hope that we can continue including this, just as I said during the original stage 3 promotion.

Reviewer: ```suggestion
JHD: I don't actually agree with one of the things on your slide show. I don't think we have to worry too much about someone making a package that does the bad thing. I think they'll do that anyway if they want to; and they probably won't regardless, but that said, I think it looks like there's more risks than having it that aren't on this slide. As you mentioned, people will reach for `.slice(-1)[0]`, or people will have to manually compute the index that they want. I think that's a risk. That's gross code that's easy to screw up that people do all over the place. Gross is of course subjective, but I hope that that's not a contentious subjective opinion. So I think it's really important to have it. Code points aren't enough to solve what people actually want; which is some form of grapheme clusters. I think that there's a very large amount of strings that aren't human sentences where surrogate pairs matter. There's all sorts of things like code enums, and plenty of ASCII strings in which code unit indexing is exactly what's desired, which is why the “slice” and “length minus 1” patterns are common. So I hope that we can continue including this, just as I said during the original stage 3 promotion.
```
